node/core/parachains-inherent/src/lib.rs

```diff
@@ -143,6 +143,52 @@ impl sp_inherents::InherentDataProvider for ParachainsInherentDataProvider {
 	}
 }
 
+/// Parachains inherent-data provider
+//#[derive(Encode, Decode, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct InherentDataProvider<C: sp_blockchain::HeaderBackend<Block>> {
```

Comment:
This should not be created.

Reply:
Why not? To save a few bytes and allocation?

Reply:
No. This type just doesn't make any sense. We already have the `InherentDataProvider`. The implementation just needs to be changed. Here you created a second type and hacked the old type into this.

Reply:
Fair. I was thinking of calling this the `ParachainsInherentDataProviderCreator` but I get where you are coming from.

Reply:
The only reason to keep the old instance and create a new one as a "provider creator" in the first place was the benchmarking code here https://github.com/paritytech/polkadot/pull/6328/files#diff-2e324183f8b01329c246bc1cf919b043150b5a88c94bf80d989a4034c0b2bd8dR372-R375. As `provide_inherent_data` assumes all the data comes from `self`, I'm having a bit of a struggle here.

Reply:
Do I create a `BenchmarkProvider` that always returns that mocked data?

Reply:
Either that or make the provider an enum. Where one variant returns the given data.